src-tauri/src/node/local_node_adapter.rs

```diff
@@ -0,0 +1,327 @@
+            tcp_listener_port,
+            use_pruned_mode: false,
+            required_initial_peers: 3,
+            use_tor: false,
+            tor_control_port: None,
+        }
+    }
+
+    pub fn get_grpc_address(&self) -> Option<(String, u16)> {
+        self.grpc_address.clone()
+    }
+
+    pub fn tcp_address(&self) -> String {
+        format!("/ip4/127.0.0.1/tcp/{}", self.tcp_listener_port)
+    }
+
+    pub fn get_service(&self) -> Option<NodeAdapterService> {
+        if let Some(grpc_address) = self.get_grpc_address() {
+            Some(NodeAdapterService::new(
+                format!("http://{}:{}", grpc_address.0, grpc_address.1),
+                self.required_initial_peers,
+            ))
+        } else {
+            None
+        }
+    }
+
+    pub fn use_tor(&mut self, use_tor: bool) {
+        self.use_tor = use_tor;
+    }
+
+    pub fn set_tor_control_port(&mut self, tor_control_port: Option<u16>) {
+        self.tor_control_port = tor_control_port;
+    }
+}
+
+#[async_trait]
+impl NodeAdapter for LocalNodeAdapter {
+    fn get_grpc_address(&self) -> Option<(String, u16)> {
+        self.get_grpc_address()
+    }
+
+    fn get_service(&self) -> Option<NodeAdapterService> {
+        self.get_service()
+    }
+
+    async fn get_connection_address(&self) -> Result<String, anyhow::Error> {
+        Ok(self.tcp_address())
+    }
+}
+
+impl ProcessAdapter for LocalNodeAdapter {
+    type StatusMonitor = NodeStatusMonitor;
+    type ProcessInstance = ProcessInstance;
+
+    #[allow(clippy::too_many_lines)]
+    fn spawn_inner(
+        &self,
+        data_dir: PathBuf,
+        _config_dir: PathBuf,
+        log_dir: PathBuf,
+        binary_version_path: PathBuf,
+        _is_first_start: bool,
+    ) -> Result<(ProcessInstance, Self::StatusMonitor), anyhow::Error> {
+        let inner_shutdown = Shutdown::new();
+
+        info!(target: LOG_TARGET, "Starting minotari node");
+        let working_dir: PathBuf = data_dir.join("node");
+        let network_dir = working_dir.join(Network::get_current().to_string().to_lowercase());
+        fs::create_dir_all(&network_dir)?;
+        let migration_file = network_dir.join("migrations.json");
+        let mut migration_info = MinotariNodeMigrationInfo::load_or_create(&migration_file)?;
+
+        if migration_info.version < 1 {
+            // Delete the peer info db.
+            let peer_db_dir = network_dir.join("peer_db");
+
+            info!(target: LOG_TARGET, "Node migration v1: removing peer db at {:?}", peer_db_dir);
+
+            if peer_db_dir.exists() {
+                let _unused = fs::remove_dir_all(peer_db_dir).inspect_err(|e| {
+                    warn!(target: LOG_TARGET, "Failed to remove peer db: {:?}", e);
+                });
+            }
+            info!(target: LOG_TARGET, "Node Migration v1 complete");
+            migration_info.version = 1;
+        }
+        migration_info.save(&migration_file)?;
+
+        // if is_first_start {
```

Comment:
🤔 ?

Reply:
https://github.com/tari-project/universe/blob/main/src-tauri/src/node_adapter.rs#L143

I have no idea, we merge it from main